File: OnlineExamSystem/StudentInfoAnalysis/QuestionInfo.cpp
#include "QuestionInfo.h"

QuestionInfo::QuestionInfo() {

}

std::string QuestionInfo::GetQuestionText() {
	return _questionText;
}

void QuestionInfo::SetQuestionText(std::string text) {
	_questionText = text;
}

std::string QuestionInfo::GetWrongAnswer0(){
	return _wrongAnswer0;
}
std::string QuestionInfo::GetWrongAnswer1() {
	return _wrongAnswer1;
}
std::string QuestionInfo::GetWrongAnswer2() {
	return _wrongAnswer2;
}
std::string QuestionInfo::GetCorrectAnswer0() {
	return _correctAnswer0;
}

void QuestionInfo::SetWrongAnswer0(std::string text) {
	_wrongAnswer0 = text;
}
void QuestionInfo::SetWrongAnswer1(std::string text) {
	_wrongAnswer1 = text;
}
void QuestionInfo::SetWrongAnswer2(std::string text) {
	_wrongAnswer2 = text;
}
void QuestionInfo::SetCorrectAnswer0(std::string text) {
	_correctAnswer0 = text;
}
int QuestionInfo::GetGlobalRightCount() {
	return _globalRightCount;
}
int QuestionInfo::GetGlobalCount() {
	return _globalCount;
}
int QuestionInfo::GetDifficultyMultiplier() {
	return _difficultyMultiplier;
}
void QuestionInfo::SetGlobalRightCount(int value) {
	_globalRightCount = value;

}
void QuestionInfo::SetGlobalCount(int value) {
	_globalCount = value;

}
void QuestionInfo::SetDifficultyMultiplier(int value) {
	_difficultyMultiplier = value;

}

File: OnlineExamSystem/ExamSystem.Core/SubModels/QuestionInfo.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ExamSystem.Core.SubModels
{
    public class QuestionInfo : QuestionInfoWrapper
    {
        public enum Difficulty { VeryEasy = 0, Easy, Medium, Hard, Extreme }

        public string QuestionText
        {
            get
            {
                return GetQuestionText(_questionInfoPointer);
            }
            set
            {
                SetQuestionText(_questionInfoPointer, value);
            }
        }

        public string WrongAnswer0
        {
            get
            {
                return GetWrongAnswer0(_questionInfoPointer);
            }
            set
            {
                SetWrongAnswer0(_questionInfoPointer, value);
            }
        }

        public string WrongAnswer1
        {
            get
            {
                return GetWrongAnswer1(_questionInfoPointer);
            }
            set
            {
                SetWrongAnswer1(_questionInfoPointer, value);
            }
        }

        public string WrongAnswer2
        {
            get
            {
                return GetWrongAnswer2(_questionInfoPointer);
            }
            set
            {
                SetWrongAnswer2(_questionInfoPointer, value);
            }
        }

        public string CorrectAnswer0
        {
            get
            {
                return GetCorrectAnswer0(_questionInfoPointer);
            }
            set
            {
                SetCorrectAnswer0(_questionInfoPointer, value);
            }
        }

        public int GlobalRightCount
        {
            get
            {
                return GetGlobalRightCount(_questionInfoPointer);
            }
            set
            {
                SetGlobalRightCount(_questionInfoPointer, value);
            }
        }

        public int GlobalCount
        {
            get
            {
                return GetGlobalCount(_questionInfoPointer);
            }
            set
            {
                SetGlobalCount(_questionInfoPointer, value);
            }
        }

        public Difficulty DifficultyMultiplier
        {
            get
            {
                return (Difficulty)GetDifficulty(_questionInfoPointer);
            }
            set
            {
                SetDifficulty(_questionInfoPointer, (int)value);
            }
        }
        public QuestionInfo()
        {
            _questionInfoPointer = CreateQuestionInfo();
            DifficultyMultiplier = 0;
            GlobalCount = 0;
            GlobalRightCount = 0;

        }

        ~QuestionInfo()
        {
            DeleteQuestionInfo(_questionInfoPointer);
        }
    }
}
